src/ros_vision_interaction/scripts/ros_reading_evaluator.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3.8
+import glob
+import os
+import pymongo
+import rospy
+import sys
+import time
+
+from std_msgs.msg import Bool
+from vision_project_tools import init_db
+from vision_project_tools import EngineStateDb as StateDb
+from controllers.vision_project_delegator import INITIAL_STATE_DB
+from reading_evaluator import ReadingEvaluator
+
+sys.path.append('/root/catkin_ws/src/ffmpeg')
+
+
+class RosReadingEvaluator:
+
+    def __init__(
+            self,
+            statedb,
+            reading_evaluator,
+            source_directory,
+            file_prefix,
+            extension="wav",
+            seconds_to_check_for_audio_files=5
+    ):
+        if not os.path.exists(source_directory):
+            raise FileNotFoundError("The upload directory \'{}\' does not exist".format(source_directory))
+
+        self._source_directory = source_directory
+        self._file_prefix = file_prefix
+        self._extension = extension
+        self._seconds_to_check_for_audio_files = seconds_to_check_for_audio_files
+
+        is_record_evaluation_topic = rospy.get_param("controllers/is_record/evaluation")
+        self._is_record_subscriber = rospy.Subscriber(
+            is_record_evaluation_topic,
+            Bool,
+            callback=self.reading_analyzer_callback,
+            queue_size=1
+        )
+
+        self._state_database = statedb
+        self._reading_evaluator = reading_evaluator
+
+    def reading_analyzer_callback(self, is_record):
+        if not is_record.data:
+            rospy.loginfo("Evaluator callback starting")
+            audio_file = self.find_audio_file(
+                self._source_directory,
+                self._extension,
+                self._file_prefix
+            )
+            reading_time = self._reading_evaluator.get_total_speaking_time(audio_file)
+            rospy.loginfo(f"Reading time: {reading_time}")
+            reading_eval_index = self._state_database.get("reading eval index")
+            try:
+                num_of_words = self._state_database.get("reading eval data")[reading_eval_index]["word count"]
+            except IndexError or KeyError:
+                rospy.logerr(f"Reading task data not found for index:{reading_eval_index}")
+                num_of_words = 0
+            try:
+                reading_speed = num_of_words / reading_time
+                rospy.loginfo("Reading speed: {}".format(reading_speed))
+            except ZeroDivisionError:
+                rospy.logerr("Reading time is 0, setting reading speed to 0")
+                reading_speed = 0
+            self._state_database.set("current eval score", reading_speed)
+
+    def find_audio_file(
```

Comment:
I'm curious about the design choice here. It seems like with this information, you should know the file is there.

Perhaps you could just check if the file exists and then read the file?

Reply:
I have this functionality in a separate method because I wasn't sure if I would need to find and analyze audio files with other file prefixes. We might still need to do that, so I'll keep the method for now.